spec/safe_statements_spec.rb

```diff
@@ -2110,6 +2110,762 @@ def up
+              ensure
+                ActiveRecord::Base.connection.schema_search_path = old_search_path
+              end
+            end
+
+            it "raises error when parent table does not exist and fully qualified name provided" do
+              migration = Class.new(migration_klass) do
+                def up
+                  unsafe_partman_create_parent "public.foos3", key: :created_at, interval: "monthly"
+                end
+              end
+
+              expect do
+                migration.suppress_messages { migration.migrate(:up) }
+              end.to raise_error(ActiveRecord::StatementInvalid, /Please create parent table first: public.foos3/)
+            end
+
+            it "raises error when template table does not exist" do
+              migration = Class.new(migration_klass) do
+                def up
+                  safe_create_partitioned_table :foos3, type: :range, key: :created_at do |t|
+                    t.timestamps :null => false
+                    t.text :text_column
+                  end
+
+                  unsafe_partman_create_parent :foos3, key: :created_at, interval: "monthly", template_table: :foos3_template
+                end
+              end
+
+              expect do
+                migration.suppress_messages { migration.migrate(:up) }
+              end.to raise_error(PgHaMigrations::InvalidMigrationError, "Could not find foos3_template in search path")
+            end
+
+            it "raises error when template table does not exist and fully qualified name provided" do
+              migration = Class.new(migration_klass) do
+                def up
+                  safe_create_partitioned_table :foos3, type: :range, key: :created_at do |t|
+                    t.timestamps :null => false
+                    t.text :text_column
+                  end
+
+                  unsafe_partman_create_parent :foos3, key: :created_at, interval: "monthly", template_table: "public.foos3_template"
+                end
+              end
+
+              expect do
+                migration.suppress_messages { migration.migrate(:up) }
+              end.to raise_error(ActiveRecord::StatementInvalid, /Please create template table first/)
+            end
+
+            it "raises error when parent table is not partitioned" do
+              migration = Class.new(migration_klass) do
+                def up
+                  safe_create_table :foos3 do |t|
+                    t.timestamps :null => false
+                    t.text :text_column
+                  end
+
+                  unsafe_partman_create_parent :foos3, key: :created_at, interval: "monthly"
+                end
+              end
+
+              expect do
+                migration.suppress_messages { migration.migrate(:up) }
+              end.to raise_error(ActiveRecord::StatementInvalid, /you must have created the given parent table as ranged \(not list\) partitioned already/)
```

Comment:
```suggestion
              end.to raise_error(ActiveRecord::StatementInvalid, /you must have created the given parent table as range \(not list\) partitioned already/)
```

Reply:
Is this one of the things we should consider validating?

Reply:
This goes back to my comment here: https://github.com/braintree/pg_ha_migrations/pull/81#discussion_r1286185575

If we do that, where does it end?

Reply:
Fair enough. I feel like there probably should be some core set, but I also haven't looked in detail at the underlying code.